context/app/routes_cells.py

```diff
@@ -124,81 +124,84 @@ def _get_cluster_cells(cells, gene, min_gene_expression):
     ...             ],
     ...             "modality": "Z",
     ...             "values": {
-    ...                 "gene": 7.0
+    ...                 "VIM": 7.0
     ...             }
-    ...         }], 'gene', 10)
+    ...         }], 'gene', 'VIM', 10)
     >>> import pprint
     >>> pprint.pprint(cells)
     [{'cluster_name': 'cluster-method-a',
       'cluster_number': '1',
-      'meets_minimum_gene_expression': True,
+      'meets_minimum_expression': True,
       'modality': 'Z'},
      {'cluster_name': 'cluster-method-b',
       'cluster_number': '1',
-      'meets_minimum_gene_expression': True,
+      'meets_minimum_expression': True,
       'modality': 'Z'},
      {'cluster_name': 'cluster-method-a',
       'cluster_number': '1',
-      'meets_minimum_gene_expression': True,
+      'meets_minimum_expression': True,
       'modality': 'Z'},
      {'cluster_name': 'cluster-method-b',
       'cluster_number': '2',
-      'meets_minimum_gene_expression': True,
+      'meets_minimum_expression': True,
       'modality': 'Z'},
      {'cluster_name': 'cluster-method-a',
       'cluster_number': '1',
-      'meets_minimum_gene_expression': False,
+      'meets_minimum_expression': False,
       'modality': 'Z'},
      {'cluster_name': 'cluster-method-b',
       'cluster_number': '1',
-      'meets_minimum_gene_expression': False,
+      'meets_minimum_expression': False,
       'modality': 'Z'}]
     '''
     cluster_cells = []
     for cell in cells:
         for cluster in cell['clusters']:
             cluster_name, cluster_number = _get_cluster_name_and_number(cluster)
-            cluster_cells.append({'modality': cell['modality'], 'cluster_name': cluster_name,
-                                  'cluster_number': cluster_number,
-                                  'meets_minimum_gene_expression':
-                                  cell['values'][gene] >= min_gene_expression})
+            cluster_cell = {'cluster_name': cluster_name,
+                            'cluster_number': cluster_number,
+                            'meets_minimum_expression':
+                            cell['values'][name] >= min_expression}
+            if query_type == 'gene':
+                cluster_cell['modality'] = cell['modality']
```

Comment:
```suggestion
                            cell['values'][name] >= min_expression,
                            'modality': cell.get('modality')}
```
I like to avoid flow-control if something else can work.